aii-pxelinux/src/test/perl/unconfigure.t

```diff
@@ -0,0 +1,68 @@
+use strict;
+use warnings;
+use Test::More;
+use Test::Quattor qw(pxelinux_base_config);
+use NCM::Component::PXELINUX::constants qw(:pxe_variants :pxe_constants);
+use NCM::Component::pxelinux;
+use CAF::FileReader;
+use CAF::Object;
+use Sys::Hostname;
+use File::Basename qw(dirname);
+use Readonly;
+use testpath;
+
+
+=pod
+
+=head1 SYNOPSIS
+
+Tests for the C<Unconfigure> method.
+
+=cut
+
+Readonly my $NBPDIR_GRUB2_VALUE => '/grub/config_files';
+Readonly my $NBPDIR_PXELINUX_VALUE => '/pxelink/config_files';
+Readonly my $FILE_INITIAL_CONTENTS => 'File has been configured';
+Readonly my $SYMLINK_INITIAL_CONTENTS => 'Symlink has been defined';
+
+$CAF::Object::NoAction = 1;
+
+our $this_app = $main::this_app;
+
+my $fh;
+
+# Define a few AII options related to Grub2 support
+# Normally done by aii-shellfe
+$this_app->{CONFIG}->define(NBPDIR_PXELINUX);
+$this_app->{CONFIG}->set(NBPDIR_PXELINUX, $NBPDIR_PXELINUX_VALUE);
+$this_app->{CONFIG}->define(NBPDIR_GRUB2);
+$this_app->{CONFIG}->set(NBPDIR_GRUB2, $NBPDIR_GRUB2_VALUE);
+
+my $testpath = testpath->new('unconfigure');
+my $comp = NCM::Component::pxelinux->new('unconfigure');
+my $cfg = get_config_for_profile('pxelinux_base_config');
+my $pxe_config = $cfg->getElement('/system/network/interfaces')->getTree();
+my $ip = $pxe_config->{eth0}->{ip};
+
+# Create a non emty file matching each PXE entries
+for my $variant (PXE_VARIANT_PXELINUX, PXE_VARIANT_GRUB2) {
+    my $config_file = NCM::Component::pxelinux::filepath($cfg,$variant);
+    my $pxe_symlink = dirname($config_file) . "/" . NCM::Component::pxelinux::hexip_filename($ip,$variant);
+    set_file_contents($config_file, $FILE_INITIAL_CONTENTS);
+    set_file_contents($pxe_symlink, $SYMLINK_INITIAL_CONTENTS);
+    # Ensure that the file contents is really instanciated and reported by get_file()
+    my $fh = CAF::FileReader->new($config_file);
+    is("$fh", $FILE_INITIAL_CONTENTS, "PXE config file $config_file has expected contents");
+    $fh = CAF::FileReader->new($pxe_symlink);
+    is("$fh", $SYMLINK_INITIAL_CONTENTS, "PXE link $pxe_symlink has expected contents");
+};
+
+$comp->Unconfigure($cfg);
+for my $variant (PXE_VARIANT_PXELINUX, PXE_VARIANT_GRUB2) {
+    my $config_file = NCM::Component::pxelinux::filepath($cfg,$variant);
+    my $pxe_symlink = dirname($config_file) . "/" . NCM::Component::pxelinux::hexip_filename($ip,$variant);
+    #is($testpath->file_exists($config_file), 0, "PXE config file $config_file removed");
```

Comment:
and finally, use `ok(! $comp->file_exist()...)`, i don't think file exits returns `0` on failure, but certainly something false.